src/check_os.sh

```diff
@@ -4,20 +4,63 @@
 _OS="Unknown"
 _DISTRO="Unknown"
 
-if [[ "$(uname)" == "Linux" ]]; then
-  _OS="Linux"
-  if command -v apt > /dev/null; then
-    _DISTRO="Ubuntu"
-  elif command -v apk > /dev/null; then
-    _DISTRO="Alpine"
+function check_os::init() {
+  if check_os::is_linux; then
+    _OS="Linux"
+    if check_os::is_ubuntu; then
+      _DISTRO="Ubuntu"
+    elif check_os::is_alpine; then
+      _DISTRO="Alpine"
+    else
+      _DISTRO="Other"
+    fi
+  elif check_os::is_macos; then
+    _OS="OSX"
+  elif check_os::is_windows; then
+    _OS="Windows"
   else
-    _DISTRO="Other"
+    _OS="Unknown"
+    _DISTRO="Unknown"
   fi
-elif [[ "$(uname)" == "Darwin" ]]; then
-  _OS="OSX"
-elif [[ "$(uname)" == *"MINGW"* ]]; then
-  _OS="Windows"
-fi
+}
+
+function check_os::is_ubuntu() {
+  command -v apt > /dev/null
+}
+
+function check_os::is_alpine() {
+  command -v apk > /dev/null
+}
+
+function check_os::is_linux() {
+  [[ "$(uname)" == "Linux" ]]
+}
+
+function check_os::is_macos() {
+  [[ "$(uname)" == "Darwin" ]]
+}
+
+function check_os::is_windows() {
+  [[ "$(uname)" == *"MINGW"* ]]
+}
+
+function check_os::is_busybox() {
+
+  case "$_DISTRO" in
+
+    "Alpine")
+        return 0
+        ;;
+    *)
+      return 1
```

Comment:
You might have notice we are using everywhere in the project checking true|false for conditional booleans. However, they are raw strings are bools are not natively supported by bash. On the contrary, native conditions in bash are `0:true,>=1:false` which is a bit odd if you come from other languages. In the end this is purely a detail implementation from the bashunit insides, so it should not matter to the external user.

I am saying this because I am wondering if there is a performance gain/benefit on using 0/1 for booleans intead of true/false in which case I would be in favor of refactoring the bools inside the project to use them. Otherwise, I would suggest keep using true/false. What do you think, @skinner-m-c ? 

> I am ok either way, I want to learn from your experience, in case you know what's better :)

Reply:
I agree that it is confusing that `0` being `true` and `1` being `false` is confusing. However it comes from Unix standard of program exit codes -- that use `0` meaning "normal exit and not errors" and  anything else is an exit where the program executed with some problems.

`true` and `false` are programs, typically installed at `/bin/true` and `/bin/false`. They are provided by `coreutils` or `busybox`. Both of these programs are very simple and are unlikely to cause performance issues. There may be more of a performance hit starting these programs than their execution. See true.c in [coreutils](https://github.com/mirror/busybox/blob/master/coreutils/true.c) and [busybox](https://github.com/coreutils/coreutils/blob/master/src/true.c) for what they actually do.  So it is probably faster to use `0` and `1` for comparisons. When `true` and `false` are returned it is the output of these programs.

However, BASH also has built-in functions and that is what is actually being used here and so `/bin/true` and `/bin/false` is not actually being run. The built-in `true` and `false' is probably marginally slower than integers (see `help` in BASH to see the list of all built-in functions).  I don't have any metric to support that idea, but it is probably too small to warrant choosing `0` over `false` for example. `false` and `true` also are more readable. 

I agree that `true` and `false` should probably be used.

Reply:
In such a case, do you mind creating a PR refactoring/changing this to use true/false then? Then we can create an [ADR](https://github.com/TypedDevs/bashunit/tree/main/adrs) to write down this as a project convention, so we can keep consistency over the project when dealing with booleans :)

Reply:
I've been thinking about this topic, and while I think it's a bit odd thinking this way if you are not familiar with sh/bash, I ended up to believe that 0 (as true) and 1 (as false) is good practice in bash itself, specially when defining conditions.

--- 

While you can do this in bash
```bash
function check_os::is_busybox() {
  case "$_DISTRO" in
    "Alpine")
        return 0
        ;;
    *)
      return 1
      ;;
  esac
}
```

but you cannot do this:
```bash
function check_os::is_busybox() {
  case "$_DISTRO" in
    "Alpine")
        return true
        ;;
    *)
      return false
      ;;
  esac
}
```

<img width="577" alt="Screenshot 2024-10-03 at 16 01 52" src="https://github.com/user-attachments/assets/5d9b0023-e5be-4653-bb81-8b91d933678b">

---

TL;DR I would be in favor of using 0 (as true) and 1 (as false) when dealing with conditionals in the project. I will create an ADR for this and refactor the project to follow this convention.

Reply:
Follow up: https://github.com/TypedDevs/bashunit/pull/346

Reply:
This also gave me the idea to add assert_true and assert_false  https://github.com/TypedDevs/bashunit/pull/350

Reply:
Glad it is inspiring some more improvements. 

`true` and `false` are build-in functions in BASH. Functions in BASH cannot be returned (though you can return a string of a function name and then invoke it with `eval`). However, the return value of the last function is returned if return has no value. So it would be a two-liner.

Reply:
Do you mind reviewing the last pr where I introduced assert_true and false? I would like your feedback:)